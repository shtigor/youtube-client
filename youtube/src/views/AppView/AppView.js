export default class AppView {
  constructor(titles) {
    this.titles = titles;
  }

  render() {
    const content = document.createElement('ul');
    content.innerHTML = this.titles.map(title => `<li>${title}</li>`).join('');

    document.body.appendChild(content);
  }
}
