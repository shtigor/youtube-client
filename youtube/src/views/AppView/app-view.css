body {
  padding: 0;
  margin: 0;
  text-align: center;
  background-color: #343d46;
}

input {
  margin-top: 10px;
  margin-bottom: 15px;
}

.items {
  display: flex;
  list-style: none;
  overflow: hidden;
  cursor: grabbing;
  margin: 10px auto;
  width: 1180px;
}

.item {
  /* display: inline-block; */
  position: relative;
  background-color: white;
  border-radius: 5px;
  margin-right: 50px;
  width: 250px;
  height: 400px;
}

.title-mod {
  position: absolute;
  display: block;
  margin: 0;
  background-color: rgba(17, 108, 74, 0.9);
  color: white;
  font-weight: bold;
  text-decoration: underline;
  top: 15%;
}

.image-mod {
  margin-top: 5px;
  max-width: 240px;
}

.channel-mod,
.date-mod,
.viewers-mod {
  position: relative;
  font-weight: 600;
  padding-bottom: 15px;
  padding-left: 60px;
  width: 150px;
}

.viewers-mod {
  padding-bottom: 0;
}

.channel-mod::before {
  content: "";
  position: absolute;
  background: url("../assets/tv.png") no-repeat;
  width: 24px;
  height: 24px;
  top: -5px;
  left: 25px;
}

.date-mod::before {
  content: "";
  position: absolute;
  background: url("../assets/calendar.png") no-repeat;
  width: 24px;
  height: 24px;
  top: -5px;
  left: 25px;
}

.viewers-mod::before {
  content: "";
  position: absolute;
  background: url("../assets/viewer.png") no-repeat;
  width: 24px;
  height: 24px;
  top: -5px;
  left: 25px;
}

.description-mod {
  text-align: left;
  margin-top: 0;
  margin-left: 5px;
  width: 240px;
}

.pagination-mod {
  display: inline-block;
  width: 25px;
  height: 25px;
  background-color: white;
  border-radius: 100%;
  text-decoration: none;
  margin-right: 5px;
  color: white;
  font-size: 12px;
  line-height: 25px;
}

.red {
  display: inline-block;
  background-color: #cb3131;
  border-radius: 100%;
  width: 21px;
  height: 21px;
  line-height: 21px;
}
