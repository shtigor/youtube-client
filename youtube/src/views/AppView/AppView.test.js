import AppView from './AppView';

describe('AppView.prototype.render', () => {
  it('Should be an instatnce of Function', () => {
    expect(AppView.prototype.render).toBeInstanceOf(Function);
  });

  it('Should be render correctly', () => {
    const context = {
      titles: [
        'Video about JS',
        'Anpther video ...',
        'I need more video',
      ],
    };

    AppView.prototype.render.call(context);

    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
