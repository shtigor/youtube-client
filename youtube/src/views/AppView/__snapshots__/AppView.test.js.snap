// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView.prototype.render Should be render correctly 1`] = `"<ul><li>Video about JS</li><li>Anpther video ...</li><li>I need more video</li></ul>"`;
